### PR TITLE
Modify eslint config and remove camelcase ignores

### PR DESCRIPTION
**Pull request by louwie17**
#### Changes proposed in this Pull Request

Update the eslint camelcase config to still enforce variables and function names a camelCase, but allow for snakecase property names, given our API data is in snake case often.
- Turn off the parameter config for camelcase and also set `ignoreGlobals` on
- Make use of the TypeScript eslint `naming-convention` config instead of eslint camelcase for TypeScript files

#### Testing instructions

- Run `npm run lint`, there should be no errors
- Add the below JS to a JS file:
```
function test_function() {
	var a_variable = 'test';
	return a_variable;
}
```
- Run `npm run lint` and see if camelcase warnings show up for the function name and variable name.
- Do a search through the repo for `camelcase` ignores (there should be none now).

*

-------------------

- [ ] Added changelog entry (or does not apply)
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

File: includes/class-wc-payment-gateway-wcpay.php [hunk 1]
```diff
@@ -910,12 +910,6 @@ public function process_payment_for_order( $cart, $payment_information, $additio
 			$charge_id     = $intent->get_charge_id();
 			$client_secret = $intent->get_client_secret();
 			$currency      = $intent->get_currency();
-
-			if ( 'requires_action' === $status &&
-				$payment_information->is_merchant_initiated() ) {
-				// Allow 3rd-party to trigger some action if needed.
-				do_action( 'woocommerce_woocommerce_payments_payment_requires_action', $order, $intent_id, $payment_method, $customer_id, $charge_id, $currency );
-			}
```
Comment (mattsherman) [inline includes/class-wc-payment-gateway-wcpay.php:918]: Why was this removed? It seems unrelated to the eslint change that this PR focuses on.

**Review by mattsherman — CHANGES_REQUESTED** [review]
I tested the eslint changes that things look correct in that regard.

I left a comment regarding some removed code that doesn't seem like it is related to this eslint change. I'd like to understand why is was removed in the context of this PR.

File: includes/class-wc-payment-gateway-wcpay.php (hunk 1 above)
Comment (louwie17) [reply includes/class-wc-payment-gateway-wcpay.php:918]: Nice catch @mattsherman, I am not a 100% sure why I had removed that, I will revert that change.

**Comment by louwie17** [comment]
> I tested the eslint changes that things look correct in that regard.
> 
> I left a comment regarding some removed code that doesn't seem like it is related to this eslint change. I'd like to understand why is was removed in the context of this PR.

Thanks for the review @mattsherman 😄 A rebase seemed to fix the change you pointed out above. Are you able to give it another look?

**Comment by mattsherman** [comment]
> Thanks for the review @mattsherman 😄 A rebase seemed to fix the change you pointed out above. Are you able to give it another look?

Great -- I'll give it another look shortly!

**Review by mattsherman — APPROVED** [review]
Nice improvement to the code maintainability, @louwie17 !

Linting works as expected. Code changes look good.
